import os
import re
#import pandas as pd
#======================================================
# Config files
#======================================================
configfile: "config.yaml"

#======================================================
# Global variables
#======================================================

RAW_DATA_DIR =config['input_dir'].rstrip("/") 
RESULTS_DIR=config['results_dir'].rstrip("/")
dir_list = ["RULES_DIR","ENVS_DIR", "ADAPTERS_DIR", "CONTAMINANTS_DIR","RAW_DATA_DIR", "QC_DIR", "CLEAN_DATA_DIR", "ASSEMBLY_DIR", "vOUT_DIR", "VIRAL_DIR", "MAPPING_DIR", "PROFILE_DIR", "MERGE_DIR"]
dir_names = ["rules", "../envs", "db/adapters", "db/contaminants" ,RAW_DATA_DIR, RESULTS_DIR + "/01_QC", RESULTS_DIR + "/02_CLEAN_DATA",RESULTS_DIR + "/03_CONTIGS", RESULTS_DIR + "/04_vOTUs", RESULTS_DIR + "/05_VIRAL_ID",RESULTS_DIR + "/06_MAPPING", "05_ANVIO_PROFILE", "06_MERGED"]
dirs_dict = dict(zip(dir_list, dir_names))
RULES_DIR = 'rules'
SAMPLING_TYPE=["sub", "tot"]
SAMPLES,=glob_wildcards(RAW_DATA_DIR + "/{sample}_" + config['forward_tag'] + ".fastq")
NANOPORE_SAMPLES=SAMPLES
CONTAMINANTS=config['contaminants_list']
NANOPORE=False
PAIRED=False
POOLED=config['nanopore_pooled']

READ_TYPES=[config['forward_tag']]

for fname in os.listdir(RAW_DATA_DIR):
	if fname.endswith(config['reverse_tag'] + '.fastq'):
		PAIRED=True
	elif fname.endswith(config['nanopore_tag'] + '.fastq'):
		NANOPORE=True
if PAIRED:
	READ_TYPES.append(config['reverse_tag'])
if POOLED == "True":
	NANOPORE_SAMPLES=config['nanopore_pooled_name']

#======================================================
# Rules
#======================================================
 
rule all:
	input:
		dirs_dict["QC_DIR"]+ "/pre_processing_multiqc_report.html",
		expand(dirs_dict["ASSEMBLY_DIR"] + "/{sample}_quast_report.{sampling}.txt", sample=SAMPLES, sampling=SAMPLING_TYPE),
		expand(dirs_dict["QC_DIR"] + "/{sample}_nanopore_report.html",sample=NANOPORE_SAMPLES),
		expand(dirs_dict["MAPPING_DIR"]+ "/high_confidence_vOTU_abundance_table.{sampling}.biom", sampling=SAMPLING_TYPE),
		expand(dirs_dict["MAPPING_DIR"]+ "/low_confidence_vOTU_abundance_table.{sampling}.biom", sampling=SAMPLING_TYPE),
		expand(dirs_dict["MAPPING_DIR"]+ "/vOTU_summary.{sampling}.txt",sampling=SAMPLING_TYPE),

include: os.path.join(RULES_DIR, 'qualityControl.smk')
include: os.path.join(RULES_DIR, 'assembly.smk')
include: os.path.join(RULES_DIR, 'vOTUclustering.smk')
include: os.path.join(RULES_DIR, 'viralFiltering.smk')
include: os.path.join(RULES_DIR, 'abundance.smk')
include: os.path.join(RULES_DIR, 'taxonomyAssignment.smk')


		#awk '{{for(x=1;x<=NF;x++)if($x~/^>/){{sub(/^>/,">orf|"++i"| ")}}}}1' {output.high_aa_temp} > {output.high_aa}
		#awk '{{for(x=1;x<=NF;x++)if($x~/^>/){{sub(/^>/,">orf|"++i"| ")}}}}1' {output.low_aa_temp} >  {output.low_aa}
		#grep ">" {output.high_aa} | awk -F "[_ ]" '{{ print substr($1,2,length($1))","$2","$13}}' > {output.high_genome_file}
		#grep ">" {output.low_aa} | awk -F "[_ ]" '{{ print substr($1,2,length($1))","$2","$13}}' > {output.low_genome_file}
		#sed 's/[ ].*$//' {output.high_aa} > {output.high_aa_temp}
		#sed 's/[ ].*$//' {output.low_aa} > {output.low_aa_temp}


