

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Cluster Execution &mdash; Mosaic 1.0 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="../_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="../" src="../_static/documentation_options.js"></script>
        <script type="text/javascript" src="../_static/jquery.js"></script>
        <script type="text/javascript" src="../_static/underscore.js"></script>
        <script type="text/javascript" src="../_static/doctools.js"></script>
    
    <script type="text/javascript" src="../_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" />
    <link rel="prev" title="Requirements" href="../getting_started/userGuide.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="../index.html" class="icon icon-home"> Mosaic
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">Getting started</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../getting_started/userGuide.html">Requirements</a></li>
<li class="toctree-l1"><a class="reference internal" href="../getting_started/userGuide.html#installation">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="../getting_started/userGuide.html#running-mosaic">Running Mosaic</a></li>
</ul>
<p class="caption"><span class="caption-text">Project Info</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="#">Cluster Execution</a></li>
<li class="toctree-l1"><a class="reference internal" href="#config-file">Config File</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#single-end">Single End</a></li>
<li class="toctree-l2"><a class="reference internal" href="#general">General</a></li>
<li class="toctree-l2"><a class="reference internal" href="#nanopore">Nanopore</a></li>
<li class="toctree-l2"><a class="reference internal" href="#trimmomatic">Trimmomatic</a></li>
<li class="toctree-l2"><a class="reference internal" href="#cleaning">Cleaning</a></li>
<li class="toctree-l2"><a class="reference internal" href="#normalization-and-subsampling">Normalization and Subsampling</a></li>
<li class="toctree-l2"><a class="reference internal" href="#contigs">Contigs</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="#outputs">Outputs</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#qc">01_QC</a></li>
<li class="toctree-l2"><a class="reference internal" href="#clean-data">2_CLEAN_DATA</a></li>
<li class="toctree-l2"><a class="reference internal" href="#id1">3_CONTIGS</a></li>
<li class="toctree-l2"><a class="reference internal" href="#votus">4_vOTUs</a></li>
<li class="toctree-l2"><a class="reference internal" href="#viral-id">5_VIRAL_ID</a></li>
<li class="toctree-l2"><a class="reference internal" href="#mapping">6_MAPPING</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="#support">Support</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">Mosaic</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../index.html">Docs</a> &raquo;</li>
        
      <li>Cluster Execution</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="../_sources/project_info/support.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="cluster-execution">
<h1>Cluster Execution<a class="headerlink" href="#cluster-execution" title="Permalink to this headline">¶</a></h1>
<p>The use of Snakemake allows Mosaic’s jobs to be executed on cluster engines with SURFsara-Grid, PBS-TORQUE or Slurm job schedulers.</p>
<p>In order to execute Mosaic in a cluster, run the following code according to your job scheduler:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">snakemake</span> <span class="o">--</span><span class="n">cluster</span> <span class="n">qsub</span> <span class="o">-</span><span class="n">j</span> <span class="mi">32</span>               <span class="c1">#PBS-TORQUE</span>
<span class="n">snakemake</span> <span class="o">--</span><span class="n">cluster</span> <span class="n">sbatch</span> <span class="o">-</span><span class="n">j</span> <span class="mi">32</span>             <span class="c1">#Slurm</span>
</pre></div>
</div>
<p>For additional information about cluster configuration, please refer to Snakemake examples
avaliable <a class="reference external" href="https://snakemake.readthedocs.io/en/stable/snakefiles/configuration.html#cluster-configuration">here</a>.</p>
</div>
<div class="section" id="config-file">
<h1>Config File<a class="headerlink" href="#config-file" title="Permalink to this headline">¶</a></h1>
<p>Many values can be modified depending on user needs. Mosaic’s JSON config file allows the user to select types of data and thresholds used for the analysis depending on their needs. This file may be used as supplementary material to replicate the analyses. Using this, plus the conda JSON files describing all dependencies versions, we expect the users to consistently retrieve the same results independent on the machine.</p>
<p>In order to change this values, you can append the desired values to the main command. For example, in order to change the minimun length of a contig to 2000bp, you can use:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>snakemake -j $nCores --use-conda --config min_len=2000 basecalled_data=$fastqDir results_dir=$reusultDir
</pre></div>
</div>
<p>Below, you can find all the variables, and their default values.</p>
<div class="section" id="single-end">
<h2>Single End<a class="headerlink" href="#single-end" title="Permalink to this headline">¶</a></h2>
<p>Mosaic automatically detects the Illumina input file. Please note that Spades does not support metagenomic assembly for SE files. Hence, Mosaic assembles SE reads using Spades single-cell flag.</p>
</div>
<div class="section" id="general">
<h2>General<a class="headerlink" href="#general" title="Permalink to this headline">¶</a></h2>
<table border="1" class="docutils">
<colgroup>
<col width="30%" />
<col width="22%" />
<col width="48%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Name</th>
<th class="head">Default</th>
<th class="head">Description</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>input_dir</td>
<td>“data/00_RAW_DATA”</td>
<td>Input folder with raw data</td>
</tr>
<tr class="row-odd"><td>results_dir</td>
<td>“data”</td>
<td>Directory for output files</td>
</tr>
<tr class="row-even"><td>operating_system</td>
<td>“linux”</td>
<td>Operating system ([linux], [macOs])</td>
</tr>
<tr class="row-odd"><td>forward_tag</td>
<td>“R1”</td>
<td>Forward tag used for Illumina reads</td>
</tr>
<tr class="row-even"><td>reverse_tag</td>
<td>“R2”</td>
<td>Reverse tag used for Illumina reads</td>
</tr>
<tr class="row-odd"><td>nanopore_tag</td>
<td>“nanopore”</td>
<td>Tag used for Nanopore files</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="nanopore">
<h2>Nanopore<a class="headerlink" href="#nanopore" title="Permalink to this headline">¶</a></h2>
<table border="1" class="docutils">
<colgroup>
<col width="30%" />
<col width="22%" />
<col width="48%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Name</th>
<th class="head">Default</th>
<th class="head">Description</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>nanopore_pooled</td>
<td>“False”</td>
<td>Nanopore run pooling</td>
</tr>
<tr class="row-odd"><td>nanopore_pooled_name</td>
<td>“raw”</td>
<td>Tag used for pooled Nanopore run</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="trimmomatic">
<h2>Trimmomatic<a class="headerlink" href="#trimmomatic" title="Permalink to this headline">¶</a></h2>
<table border="1" class="docutils">
<colgroup>
<col width="30%" />
<col width="22%" />
<col width="48%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Name</th>
<th class="head">Default</th>
<th class="head">Description</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>trimmomatic_leading</td>
<td>“15”</td>
<td>Leading minimum base quality</td>
</tr>
<tr class="row-odd"><td>trimmomatic_trailing</td>
<td>“15”</td>
<td>Trailing minimum base quality</td>
</tr>
<tr class="row-even"><td>trimmomatic_window_size</td>
<td>“4”</td>
<td>Window size</td>
</tr>
<tr class="row-odd"><td>trimmomatic_window_quality</td>
<td>“15”</td>
<td>Window minimum quality</td>
</tr>
<tr class="row-even"><td>trimmomatic_minlen</td>
<td>“50”</td>
<td>Read minimum length</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="cleaning">
<h2>Cleaning<a class="headerlink" href="#cleaning" title="Permalink to this headline">¶</a></h2>
<table border="1" class="docutils">
<colgroup>
<col width="30%" />
<col width="22%" />
<col width="48%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Name</th>
<th class="head">Default</th>
<th class="head">Description</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>contaminants_list</td>
<td>“GCF_000819615”</td>
<td>List of contaminants RefSeq accesion IDs
Eg: “GCF_000819615 GCA_000929915.1”</td>
</tr>
<tr class="row-odd"><td>adapters_file</td>
<td>“NexteraPE-PE.fa”</td>
<td>Adapter file for barcode removal
Please note that if you use other adapters
you need to add them to pipes/db/adapters</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="normalization-and-subsampling">
<h2>Normalization and Subsampling<a class="headerlink" href="#normalization-and-subsampling" title="Permalink to this headline">¶</a></h2>
<table border="1" class="docutils">
<colgroup>
<col width="30%" />
<col width="22%" />
<col width="48%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Name</th>
<th class="head">Default</th>
<th class="head">Description</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>min_norm</td>
<td>“2”</td>
<td>Minimum normalization coverage</td>
</tr>
<tr class="row-odd"><td>max_norm</td>
<td>“100”</td>
<td>Maximum normalization coverage</td>
</tr>
<tr class="row-even"><td>max_subsample</td>
<td>“20000000”</td>
<td>Maximum number of reads for subsampling</td>
</tr>
</tbody>
</table>
</div>
<div class="section" id="contigs">
<h2>Contigs<a class="headerlink" href="#contigs" title="Permalink to this headline">¶</a></h2>
<table border="1" class="docutils">
<colgroup>
<col width="30%" />
<col width="22%" />
<col width="48%" />
</colgroup>
<thead valign="bottom">
<tr class="row-odd"><th class="head">Name</th>
<th class="head">Default</th>
<th class="head">Description</th>
</tr>
</thead>
<tbody valign="top">
<tr class="row-even"><td>min_len</td>
<td>“1000”</td>
<td>Minimum contig length</td>
</tr>
<tr class="row-odd"><td>min_cov</td>
<td>“2”</td>
<td>Minimum contig coverage</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="section" id="outputs">
<h1>Outputs<a class="headerlink" href="#outputs" title="Permalink to this headline">¶</a></h1>
<div class="section" id="qc">
<h2>01_QC<a class="headerlink" href="#qc" title="Permalink to this headline">¶</a></h2>
<p>MultiQC report of Illumina pre-processing fastq files:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">pre_processing_multiqc_report</span><span class="o">.</span><span class="n">html</span>
</pre></div>
</div>
<p>Nanopore report of each nanopore fastq file:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">{</span><span class="n">sample</span><span class="p">}</span><span class="n">_nanopore_report</span><span class="o">.</span><span class="n">html</span>
</pre></div>
</div>
</div>
<div class="section" id="clean-data">
<h2>2_CLEAN_DATA<a class="headerlink" href="#clean-data" title="Permalink to this headline">¶</a></h2>
<p>Clean Illumina fastq files (not normalized) for subsampled and total reads:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">{</span><span class="n">sample</span><span class="p">}</span><span class="n">_unpaired_clean</span><span class="o">.</span><span class="n">sub</span><span class="o">.</span><span class="n">fastq</span>
<span class="p">{</span><span class="n">sample</span><span class="p">}</span><span class="n">_unpaired_clean</span><span class="o">.</span><span class="n">tot</span><span class="o">.</span><span class="n">fastq</span>
</pre></div>
</div>
<p>Clean Nanopore fastq files (not normalized) for subsampled and total reads:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">{</span><span class="n">sample</span><span class="p">}</span><span class="n">_clean</span><span class="o">.</span><span class="n">sub</span><span class="o">.</span><span class="n">fastq</span>
<span class="p">{</span><span class="n">sample</span><span class="p">}</span><span class="n">_clean</span><span class="o">.</span><span class="n">tot</span><span class="o">.</span><span class="n">fastq</span>
</pre></div>
</div>
</div>
<div class="section" id="id1">
<h2>3_CONTIGS<a class="headerlink" href="#id1" title="Permalink to this headline">¶</a></h2>
<p>Assembled Spades scaffolds:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">{</span><span class="n">sample</span><span class="p">}</span><span class="n">_spades_filtered_scaffolds</span><span class="o">.</span><span class="n">sub</span><span class="o">.</span><span class="n">fasta</span>
<span class="p">{</span><span class="n">sample</span><span class="p">}</span><span class="n">_spades_filtered_scaffolds</span><span class="o">.</span><span class="n">tot</span><span class="o">.</span><span class="n">fasta</span>
</pre></div>
</div>
<p>Assembled Canu scaffolds, error corrected with Pilon:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">{</span><span class="n">sample</span><span class="p">}</span><span class="n">_canu_filtered_scaffolds</span><span class="o">.</span><span class="n">sub</span><span class="o">.</span><span class="n">fasta</span>
<span class="p">{</span><span class="n">sample</span><span class="p">}</span><span class="n">_canu_filtered_scaffolds</span><span class="o">.</span><span class="n">tot</span><span class="o">.</span><span class="n">fasta</span>
</pre></div>
</div>
<p>Quast report of (Spades) or (Spades + Canu)  assembly:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">{</span><span class="n">sample</span><span class="p">}</span><span class="n">_quast_report</span><span class="o">.</span><span class="n">sub</span><span class="o">.</span><span class="n">txt</span>
<span class="p">{</span><span class="n">sample</span><span class="p">}</span><span class="n">_quast_report</span><span class="o">.</span><span class="n">tot</span><span class="o">.</span><span class="n">txt</span>
</pre></div>
</div>
</div>
<div class="section" id="votus">
<h2>4_vOTUs<a class="headerlink" href="#votus" title="Permalink to this headline">¶</a></h2>
<p>All assembled scaffolds:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">merged_scaffolds</span><span class="o">.</span><span class="n">sub</span><span class="o">.</span><span class="n">fasta</span>
<span class="n">merged_scaffolds</span><span class="o">.</span><span class="n">tot</span><span class="o">.</span><span class="n">fasta</span>
</pre></div>
</div>
<p>Representatives contigs after clustering:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">merged_scaffolds</span><span class="o">.</span><span class="n">sub_95</span><span class="o">-</span><span class="mf">80.</span><span class="n">fna</span>
<span class="n">merged_scaffolds</span><span class="o">.</span><span class="n">tot_95</span><span class="o">-</span><span class="mf">80.</span><span class="n">fna</span>
</pre></div>
</div>
</div>
<div class="section" id="viral-id">
<h2>5_VIRAL_ID<a class="headerlink" href="#viral-id" title="Permalink to this headline">¶</a></h2>
<p>High confidence viral contigs:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">high_confidence</span><span class="o">.</span><span class="n">sub</span><span class="o">.</span><span class="n">fasta</span>
<span class="n">high_confidence</span><span class="o">.</span><span class="n">tot</span><span class="o">.</span><span class="n">fasta</span>
</pre></div>
</div>
<p>Low confidence viral contigs:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">low_confidence</span><span class="o">.</span><span class="n">sub</span><span class="o">.</span><span class="n">fasta</span>
<span class="n">low_confidence</span><span class="o">.</span><span class="n">tot</span><span class="o">.</span><span class="n">fasta</span>
</pre></div>
</div>
<p>High confidence open reading frames (ORFs):</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">high_confidence_ORFs</span><span class="o">.</span><span class="n">sub</span><span class="o">.</span><span class="n">fasta</span>
<span class="n">high_confidence_ORFs</span><span class="o">.</span><span class="n">tot</span><span class="o">.</span><span class="n">fasta</span>
</pre></div>
</div>
<p>Low confidence open reading frames (ORFs):</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">low_confidence_ORFs</span><span class="o">.</span><span class="n">sub</span><span class="o">.</span><span class="n">fasta</span>
<span class="n">low_confidence_ORFs</span><span class="o">.</span><span class="n">tot</span><span class="o">.</span><span class="n">fasta</span>
</pre></div>
</div>
</div>
<div class="section" id="mapping">
<h2>6_MAPPING<a class="headerlink" href="#mapping" title="Permalink to this headline">¶</a></h2>
<p>Sorted bam files of sample clean reads mapped to high confidence contigs:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">{</span><span class="n">sample</span><span class="p">}</span><span class="n">_high_confidence_sorted</span><span class="o">.</span><span class="n">sub_filtered</span><span class="o">.</span><span class="n">bam</span>
<span class="p">{</span><span class="n">sample</span><span class="p">}</span><span class="n">_high_confidence_sorted</span><span class="o">.</span><span class="n">tot_filtered</span><span class="o">.</span><span class="n">bam</span>
</pre></div>
</div>
<p>Sorted bam files of sample clean reads mapped to low confidence contigs:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">{</span><span class="n">sample</span><span class="p">}</span><span class="n">_low_confidence_sorted</span><span class="o">.</span><span class="n">sub_filtered</span><span class="o">.</span><span class="n">bam</span>
<span class="p">{</span><span class="n">sample</span><span class="p">}</span><span class="n">_low_confidence_sorted</span><span class="o">.</span><span class="n">tot_filtered</span><span class="o">.</span><span class="n">bam</span>
</pre></div>
</div>
<p>Indexed bam files of sample clean reads mapped to high confidence contigs:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">{</span><span class="n">sample</span><span class="p">}</span><span class="n">_high_confidence_sorted</span><span class="o">.</span><span class="n">sub_filtered</span><span class="o">.</span><span class="n">bam</span><span class="o">.</span><span class="n">bai</span>
<span class="p">{</span><span class="n">sample</span><span class="p">}</span><span class="n">_high_confidence_sorted</span><span class="o">.</span><span class="n">tot_filtered</span><span class="o">.</span><span class="n">bam</span><span class="o">.</span><span class="n">bai</span>
</pre></div>
</div>
<p>Indexed bam files of sample clean reads mapped to low confidence contigs:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">{</span><span class="n">sample</span><span class="p">}</span><span class="n">_low_confidence_sorted</span><span class="o">.</span><span class="n">sub_filtered</span><span class="o">.</span><span class="n">bam</span><span class="o">.</span><span class="n">bai</span>
<span class="p">{</span><span class="n">sample</span><span class="p">}</span><span class="n">_low_confidence_sorted</span><span class="o">.</span><span class="n">tot_filtered</span><span class="o">.</span><span class="n">bam</span><span class="o">.</span><span class="n">bai</span>
</pre></div>
</div>
<p>Abundance tables in txt format:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">high_confidence_vOTU_abundance_table</span><span class="o">.</span><span class="n">sub</span><span class="o">.</span><span class="n">txt</span>
<span class="n">high_confidence_vOTU_abundance_table</span><span class="o">.</span><span class="n">tot</span><span class="o">.</span><span class="n">txt</span>
<span class="n">low_confidence_vOTU_abundance_table</span><span class="o">.</span><span class="n">sub</span><span class="o">.</span><span class="n">txt</span>
<span class="n">low_confidence_vOTU_abundance_table</span><span class="o">.</span><span class="n">tot</span><span class="o">.</span><span class="n">txt</span>
</pre></div>
</div>
<p>Abundance tables in BIOM JSON format:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">high_confidence_vOTU_abundance_table_json</span><span class="o">.</span><span class="n">sub</span><span class="o">.</span><span class="n">biom</span>
<span class="n">high_confidence_vOTU_abundance_table_json</span><span class="o">.</span><span class="n">tot</span><span class="o">.</span><span class="n">biom</span>
<span class="n">low_confidence_vOTU_abundance_table_json</span><span class="o">.</span><span class="n">sub</span><span class="o">.</span><span class="n">biom</span>
<span class="n">low_confidence_vOTU_abundance_table_json</span><span class="o">.</span><span class="n">tot</span><span class="o">.</span><span class="n">biom</span>
</pre></div>
</div>
<p>Contigs summary table:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">vOTU_summary</span><span class="o">.</span><span class="n">txt</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="support">
<h1>Support<a class="headerlink" href="#support" title="Permalink to this headline">¶</a></h1>
<p>The easiest way to get help with the project is to send an email to <a class="reference external" href="mailto:lm&#46;forero10&#37;&#52;&#48;uniandes&#46;edu&#46;co">lm<span>&#46;</span>forero10<span>&#64;</span>uniandes<span>&#46;</span>edu<span>&#46;</span>co</a>.
The other good way is to open an issue on Mosaic’s Github page at <a class="reference external" href="https://github.com/lauramilena3/Mosaic/issues">https://github.com/lauramilena3/Mosaic/issues</a>.</p>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
      
        <a href="../getting_started/userGuide.html" class="btn btn-neutral float-left" title="Requirements" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2019, Laura M Forero

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>